keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -115,24 +116,29 @@ class ArtifactListener(
                 // todo eb: is there a better way to think of docker status?
                 docker -> null
               }
-              artifactRepository.store(artifact.name, artifact.type, latestVersion, status)
+              repository.storeArtifact(artifact.name, artifact.type, latestVersion, status)
             }
           }
         }
       }
     }
 
   private fun getLatestStoredVersion(artifact: DeliveryArtifact): String? =
-    artifactRepository.versions(artifact).sortedWith(artifact.versioningStrategy.comparator).firstOrNull()
+    repository.artifactVersions(artifact).sortedWith(artifact.versioningStrategy.comparator).firstOrNull()
 
   private suspend fun getLatestDeb(artifact: DebianArtifact): String? =
     artifactService.getVersions(artifact.name).firstOrNull()
 
-  private suspend fun getLatestDockerTag(artifact: DockerArtifact): String? = clouddriverService
-    .findDockerTagsForImage("*", artifact.name)
-    .distinct()
-    .sortedWith(artifact.versioningStrategy.comparator)
-    .firstOrNull()
+  private suspend fun getLatestDockerTag(artifact: DockerArtifact): String? {
+    val serviceAccount = artifact.deliveryConfigName?.let { repository.getDeliveryConfig(it) }
```

Comment:
I went through all this trouble to get the delivery config service account here, but now that I look at it I think this qualifies as a backend call that should be using keel's identity, not the user's service account. Let me know, I'd be happy to undo it.

Reply:
Hmm. It looks like this function is used when we periodically go through all artifacts and make sure the latest version is stored for them. I'd say it is good to pull the correct service account here.